docs/tutorial/03-resources.md

```diff
@@ -19,581 +19,592 @@ socialImageTitle: Cadence Resources
-        "HelloAssetTutorial": {
-            "Fields": [
-                39
-            ],
-            "ResourceType": {
-                "Fields": [
-                    {
-                        "Identifier": "uuid",
-                        "Type": {}
-                    }
-                ],
-                "Initializers": null,
-                "Location": {
-                    "Address": "0x0000000000000005",
-                    "Name": "HelloWorld",
-                    "Type": "AddressLocation"
+    "value": [
+        {
+            "key": {
+                "value": "value",
+                "type": "String"
+            },
+            "value": {
+                "value": {
+                    "id": "A.0000000000000006.HelloResource.HelloAsset",
+                    "fields": [
+                        {
+                            "value": {
+                                "value": "269380348805120",
+                                "type": "UInt64"
+                            },
+                            "name": "uuid"
+                        }
+                    ]
+                },
+                "type": "Resource"
+            }
+        },
+        {
+            "key": {
+                "value": "type",
+                "type": "String"
+            },
+            "value": {
+                "value": "A.0000000000000006.HelloResource.HelloAsset",
+                "type": "String"
+            }
+        },
+        {
+            "key": {
+                "value": "path",
+                "type": "String"
+            },
+            "value": {
+                "value": {
+                    "domain": "storage",
+                    "identifier": "HelloAssetTutorial"
                 },
-                "QualifiedIdentifier": "HelloWorld.HelloAsset"
+                "type": "Path"
             }
         }
-    }
+    ],
+    "type": "Dictionary"
 }
 ```
 
-You'll also see `FlowToken` objects. Every account is automatically initialized
-with the ability to use FlowToken assets. You don't have to worry about those for now.
+You'll also see `FlowToken` objects, and the `HelloResource` Contract.
 
-### Load Hello Transaction
+:::info[Action]
 
-Now we're going to use a transaction to call the `hello()` method from the `HelloAsset` resource.
+Run the transaction from account `0x07` and compare the differences between the accounts.
+
+:::
+
+### Checking for Existing Storage
+
+In real applications, you need to check the location path you are storing in to make sure both cases are handled properly.
 
 :::info[Action]
 
-Open the transaction named `Load Hello`.
+First, update the authorization [entitlement] in the prepare phase to include `BorrowValue`:
 
-`Load Hello` should contain the following code:
+:::
+
+```cadence
+prepare(acct: auth(BorrowValue, SaveValue) &Account) {
+    // Existing code...
+}
+```
+
+:::info[Action]
+
+Next, add a `transaction`-level variable to store a result `String`:
```

Comment:
I'm worried that learners will not know what a transaction-level variable is here and won't know what to do or where to put it